ribs/visualize.py

```diff
@@ -453,3 +453,145 @@ def sliding_boundaries_archive_heatmap(archive,
 
     # Create the colorbar.
     ax.figure.colorbar(t, ax=ax, pad=0.1)
+
+
+def parallel_axes_plot(archive,
+                       ax=None,
+                       bc_order=None,
+                       axis_labels=None,
+                       cmap="magma",
+                       linewidth=1.5,
+                       alpha=0.8,
+                       vmin=None,
+                       vmax=None):
+    """Plots a parallel axes plot of an archive with an N-Dimensional behavior
+    space.
+
+    This visualization is meant to see the coverage of the behavior space at a
+    glance. It is possible to glean the correlations between consecutive axes,
+    but the results should be interpreted carefully.
+
+    Examples:
+        .. plot::
+            :context: close-figs
+
+            >>> import numpy as np
+            >>> import matplotlib.pyplot as plt
+            >>> from ribs.archives import GridArchive
+            >>> from ribs.visualize import grid_archive_heatmap
+            >>> # Populate the archive with the negative sphere function.
+            >>> archive = GridArchive([20, 20, 20], [(-1, 1), (-1, 1), (-1, 1)])
+            >>> archive.initialize(solution_dim=2)
+            >>> for x in np.linspace(-1, 1, 100):
+            ...     for y in np.linspace(-1, 1, 100):
+            ...         for z in np.linspace(-1, 1, 100):
+            ...             archive.add(solution=np.array([x,y,z]),
+            ...                         objective_value=-(x**2 + y**2 + z**2),
+            ...                         behavior_values=np.array([x,y,z]))
+            >>> # Plot a heatmap of the archive.
+            >>> plt.figure(figsize=(8, 6))
+            >>> parallel_axes_plot(archive)
+            >>> plt.title("Negative sphere function")
+            >>> plt.ylabel("axis values")
+            >>> plt.show()
+
+    Args:
+        archive (SlidingBoundariesArchive): A 2D SlidingBoundariesArchive.
+        ax (matplotlib.axes.Axes): Axes on which to plot the heatmap. If None,
+            the current axis will be used.
+        bc_order (list<str>): By default, the BCs of the parallel axes will
```

Comment:
I think most people will not want to pass in `behavior_0`, ... Perhaps we can just make it a list of int?

Reply:
Actually, I think we may want to combine this with `axis_labels`, as it can be a bit confusing. Say the user passes in `bc_order` of `[2,0,5]`. Now they pass in `axis_labels` of `["a", "b", "c"]`. They might expect that "a" goes to 2 and "b" goes to 0 and "c" goes to 5, but based on my current understanding, "a" would go to 0 and "b" would go to 2.

To combine, I think we could have each entry be either an int or a tuple. If an int, it is just the BC order. If a tuple, it contains an int with the BC index followed by a string with the axis name.